kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -1265,6 +1332,31 @@ class ResidualBasedNewtonRaphsonStrategy
 
     bool mKeepSystemConstantDuringIterations; // Flag to allow keeping system matrix constant during iterations
 
+    /**
+     * @brief This vector stores the non-converged solutions at each iteration
+     * @details Each entry in the vector represents the solution vector at a given iteration which did not meet the convergence criteria.
+     */
+    std::vector<Vector> NonconvergedSolutions;
+
+    /**
+     * @brief This matrix stores the non-converged solutions
+     * @details The matrix is structured such that each column represents the solution vector at a specific non-converged iteration.
+     */
+    Matrix mNonconveregedSolutionsMatrix;
+
+    /**
+     * @brief Flag indicating whether to store non-converged solutions
+     * @details Only when set to true (by calling the SetUpNonconvergedSolutionsGathering method) will the non-converged solutions at each iteration be stored.
+     */
+    bool mStoreNonconvergedSolutionsFlag = false;
+
+    /**
+     * @brief The number of iterations performed
+     * @details This variable stores the count of iterations performed during the solution process.
+     */
+    unsigned int mNumberOfIterations;
```

Comment:
you don't need this as you are already storing it in 

~~~cpp
r_model_part.GetProcessInfo()[NL_ITERATION_NUMBER] = iteration_number;
~~~